test/utils/typed_value.hpp

```diff
@@ -0,0 +1,29 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+#pragma once
+
+#include "types.hpp"
+#include "value.hpp"
+
+namespace fizzy::test
+{
+struct TypedValue
+{
+    Value value;
+    ValType type;
+
+    /// Assign type to the typeless Value.
+    /// No validation is possible, so the correctness is on the user only.
+    constexpr TypedValue(ValType ty, Value v) noexcept : value{v}, type{ty} {}
+
+    constexpr TypedValue(int32_t v) noexcept : TypedValue{ValType::i32, v} {}
```

Comment:
This is in my options minimal set of constructors. Problem here is that on macOS `uint64_t` is defined differently than on Linux. On macOS the literals of `long` type are ambiguous, e.g. `TypedValue(0x100000000)` and it must be explicitly converted to `uint64_t` as `TypedValue(0x100000000_u64)`.